crates/shim/src/lib.rs

```diff
@@ -171,9 +182,18 @@ pub fn socket_address(socket_path: &str, namespace: &str, id: &str) -> String {
         hasher.finish()
     };
 
-    format!("unix://{}/{:x}.sock", SOCKET_ROOT, hash)
+    #[cfg(unix)]
+    {
+        format!("unix://{}/{:x}.sock", SOCKET_ROOT, hash)
+    }
+
+    #[cfg(windows)]
+    {
+        format!(r"\\.\pipe\containerd-shim-{}-pipe", hash)
+    }
```

Comment:
nit: if/else might be better here:

Like:
```rust
let machine_kind = if cfg!(unix) {
  "unix"
} else if cfg!(windows) {
  "windows"
} else {
  "unknown"
};
```

Reply:
and return the formatted strings in the if/else branches?

Reply:
Right, this will be a compile time if/else branches.

Reply:
Updated